signer/src/block_observer.rs

```diff
@@ -380,17 +380,23 @@ impl<C: Context, B> BlockObserver<C, B> {
     async fn process_stacks_blocks(&self) -> Result<(), Error> {
         tracing::info!("processing stacks block");
         let stacks_client = self.context.get_stacks_client();
+        let db = self.context.get_storage_mut();
```

Comment:
I suspect you don't need mut here

Reply:
We do for the call to `write_stacks_block_headers` below.